Sources/NIOCore/ByteBuffer-core.swift

```diff
@@ -884,6 +884,34 @@ public struct ByteBuffer {
         }
         return true
     }
+    
+    /// The `ByteBuffer` will successfully be shrunk if the requested capacity is less than the current capacity, 
+    /// and the requested capacity is more than the number of readable bytes in the buffer.
+    /// If either condition is not true, the buffer will not be shrunk.
+    ///
+    /// - Parameter desiredCapacity: The desired capacity for the buffers capacity to be shrunken to
+    /// - Returns: Bool indicating whether the buffer capacity has been shrunk to the desiredCapacity.
+    @inlinable
+    @discardableResult public mutating func shrinkBufferCapacity(to desiredCapacity: Int) -> Bool {
+        let desiredCapacity = ByteBuffer.addPaddingTo(desiredCapacity)
+        guard desiredCapacity < capacity, desiredCapacity > readableBytes else {
```

Comment:
```suggestion
        guard desiredCapacity < self.capacity, desiredCapacity > self.readableBytes else {
```

missing `self.`s make this very difficult to review for correctness. Mind adding those?

Reply:
Noted.